crates/agent-sql/src/connector_tags.rs

```diff
@@ -42,6 +42,47 @@ pub async fn dequeue(txn: &mut sqlx::Transaction<'_, sqlx::Postgres>) -> sqlx::R
     .await
 }
 
+/// JobStatus is the possible outcomes of a handled connector tag.
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase", tag = "type")]
+pub enum JobStatus {
+    Queued,
+    PullFailed,
+    SpecFailed,
+    OpenGraphFailed { error: String },
+    Success,
+}
+
+// ConnectorTags represent all of the tags defined for a particular connector
+#[derive(Debug, FromRow)]
```

Comment:
Right, I think I have an alternative proposal for how to go about this:

First, all we really need to validate is that the connector _image_ is defined in the connectors table. The image alone is what implies ownership, and we very frequently are using alternative or draft tags of images, so it would be a big UX issue even just for us if we enforced tag equivalence.

Second, rather than loading connectors and tags, you can frame the whole thing as a query akin to `resolve_expanded_spec_rows` like:

```
pub async fn resolve_unknown_connectors(
    seed_ids: Vec<Id>,
    txn: &mut sqlx::Transaction<'_, sqlx::Postgres>,
) -> sqlx::Result<Vec<UnknownConnector>> {
```

Where UnknownConnector is a struct with the offending spec catalog_name and the connector.

This would be run _after_ `apply_updates_for_row` has run for all specs, such that the `live_specs` table has already been updated with the provisional connector. See how we run `expanded_specifications` for an example of another place we're using this "query after you've already updated" technique.

What's nice about this approach is that the entire check can be framed as a single SQL query in a stand-alone function which returns zero or more rows, and any row > zero Is mapped into an error.

When we're ready to remove this check, we remove the function and its call.

Note this approach is invariant to whether we're checking images and tags, or just images (but again, we should just check images).

Reply:
Done!